Dockerfile

```diff
@@ -8,5 +8,28 @@
 # TARGETARCH
 FROM docker.io/library/alpine:3.18.2@sha256:82d1e9d7ed48a7523bdebc18cf6290bdb97b82302a8a9c27d4fe885949ea94d1
 
-RUN apk add --no-cache curl iputils bind-tools tcpdump
+RUN apk add --no-cache curl iputils bind-tools tcpdump tar
+
+# INSTALL dnsperf and its dependencies
+RUN apk update \
+    && apk del libressl-dev \
+    && apk add openssl-dev ldns-dev ck-dev\
+    && apk add pkgconfig build-base gcc libtool
+WORKDIR /usr/bin
+
+# Install nghttp2 dependency
+RUN curl -L -O https://github.com/nghttp2/nghttp2/releases/download/v1.54.0/nghttp2-1.54.0.tar.gz \
+    && tar zxvf nghttp2-1.54.0.tar.gz \
+    && cd nghttp2-1.54.0 \
+    &&  ./configure \
+    && make install
+# Install dnsperf
+RUN curl -L https://www.dns-oarc.net/files/dnsperf/dnsperf-2.13.0.tar.gz --output dnsperf-2.13.0.tar.gz \
```

Comment:
same as above i.e. version arg and removing temp files.

Reply:
Fixed the args! Should I remove temp files even I have a multi stage?